pkg/prometheus/promcfg.go

```diff
@@ -523,7 +541,15 @@ func (cg *configGenerator) generatePodMonitorConfig(version semver.Version, m *v
 	return cfg
 }
 
-func (cg *configGenerator) generateServiceMonitorConfig(version semver.Version, m *v1.ServiceMonitor, ep v1.Endpoint, i int, apiserverConfig *v1.APIServerConfig, basicAuthSecrets map[string]BasicAuthCredentials) yaml.MapSlice {
+func (cg *configGenerator) generateServiceMonitorConfig(
+	version semver.Version,
+	m *v1.ServiceMonitor,
+	ep v1.Endpoint,
+	i int,
+	apiserverConfig *v1.APIServerConfig,
+	basicAuthSecrets map[string]BasicAuthCredentials,
+	bearerTokens map[string]BearerToken,
+) yaml.MapSlice {
```

Comment:
nit: this method is getting slowly out of hand, how about adding functional options a la https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis ?

Reply:
the above comment still holds, although we can address this (please) as a follow-up.